### PR TITLE
Use different list switch button for associations in views than for lists

### PR DESCRIPTION
**Pull request by brunobesson**
* better visuals (IMHO)
* prepare for a possible actions toolbar
* some ng-show -> ng-if optimisations

c2c9
c2c11

### CODE REVIEW

File: c2corg_ui/static/js/listswitchaction.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  /**
+   * @type {app.ListSwitch}
+   */
+  this.listSwitchService = appListSwitch;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.showList = this.listSwitchService.isList();
```
Comment (asaunier) [inline c2corg_ui/static/js/listswitchaction.js]: ``isList()`` does not return anything!?

File: c2corg_ui/static/js/listswitchbutton.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+goog.provide('app.ListSwitchButtonController');
```
Comment (asaunier) [inline c2corg_ui/static/js/listswitchbutton.js]: These directive/controller seem the same than the ones of the action. Perhaps you could avoid duplicating code by switching the template depending on some parameter? See for instance https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/static/js/card.js#L24-L25

Or you could group both templates into one and use ``ng-if`` to figure out which layout to use?

File: c2corg_ui/static/js/listswitchaction.js (hunk 1 above)
Comment (brunobesson) [inline c2corg_ui/static/js/listswitchaction.js]: isList() returns whether the display is configured for lists or cards: https://github.com/c2corg/v6_ui/pull/1713/files#diff-1cd444cb2344de44c8f6791257e43f67R37

Or maybe I didn't understand your question?

File: c2corg_ui/static/js/listswitchbutton.js (hunk 2 above)
Comment (brunobesson) [inline c2corg_ui/static/js/listswitchbutton.js]: Indeed I didn't know how to optimize this redundancy. If so, maybe the service itself is useless then, and I could make a much simpler PR? What do you think in terms of modeling?

File: c2corg_ui/static/js/listswitchaction.js (hunk 1 above)
Comment (asaunier) [inline c2corg_ui/static/js/listswitchaction.js]: Forget my comment, I think I have misread something :P

**Pull request closed without merging**